pkg/server/ca/manager_test.go

```diff
@@ -133,24 +138,34 @@ func (m *ManagerTestSuite) TestPersistence() {
 	current1 := m.m.getCurrentKeypairSet()
 
 	// "reload" the manager and assert the keypairs are the same
-	m.m = NewManager(m.m.c)
+	m.newManager()
 	m.Require().NoError(m.m.Initialize(ctx))
 	current2 := m.m.getCurrentKeypairSet()
 	m.requireKeypairSetKeysEqual(current1, current2)
 
 	// drop the keys, "reload" the manager, and assert the keypairs are new
 	m.catalog.SetKeyManagers(memory.New())
-	m.m = NewManager(m.m.c)
+	m.newManager()
 	m.Require().NoError(m.m.Initialize(ctx))
 	current3 := m.m.getCurrentKeypairSet()
 	m.requireKeypairSetKeysNotEqual(current2, current3)
 
 	// load the old keys, "reload" the manager, and assert the keypairs are new
 	m.catalog.SetKeyManagers(m.keymanager)
-	m.m = NewManager(m.m.c)
+	m.newManager()
 	m.Require().NoError(m.m.Initialize(ctx))
 	current4 := m.m.getCurrentKeypairSet()
 	m.requireKeypairSetKeysNotEqual(current3, current4)
+
+	// rotate the keypairs, "reload" the manager, and make sure current is persisted.
+	m.setTime(activationThreshold(current4.x509CA.cert).Add(time.Second))
```

Comment:
Might be good to have a test case here that comes in between prepare and activate
1. Advance past preparation threshold
1. "reload" and assert current + next is accurate
1. Advance past activation threshold
1. "reload" and assert that current is accurate (and maybe that next is nil?)

Reply:
sure. sounds good.